src/_c_internal_utils.c

```diff
@@ -1,11 +1,65 @@
 #define PY_SSIZE_T_CLEAN
 #include <Python.h>
+#ifdef __linux__
+#include <dlfcn.h>
+#endif
 #ifdef _WIN32
 #include <Objbase.h>
 #include <Shobjidl.h>
 #include <Windows.h>
 #endif
 
+static PyObject* mpl_display_is_valid(PyObject* module)
+{
+#ifdef __linux__
+    void* libX11;
+    // The getenv check is redundant but helps performance as it is much faster
+    // than dlopen().
+    if (getenv("DISPLAY")
+        && (libX11 = dlopen("libX11.so", RTLD_LAZY))) {
```

Comment:
This only works if you have the development package of X11 installed; it would have to check 
```suggestion
        && (libX11 = dlopen("libX11.so.6", RTLD_LAZY))) {
```
for most current distros.